### ISSUE TITLE
Share Lemmy-posts in groups

### ISSUE THREAD

**Issue by forenta** [friendica/friendica#13214]
### Is the feature request related to a problem? Please describe.

Posts from Lemmy-users in Friendica-groups are invisible to Friendica-users

### Describe the feature you'd like

If you create a post addressed to a Friendica group, it will be marked with ! or @ and then shared by the group. This behavior is not usual with Lemmy (and I assume that the problem affects kbin as well): There you open the corresponding group (community, magazine, bla) and create a post in it.
As a Friendica user, however, I don't receive this post in my timeline - because it's not obvious for Lemmy users to mention the group in which he writes.
Consequence: New posts in Friendica groups of Lemmy users are accessible if you explicitly search for them via URL - but as a member you don't get to know about their existence at all.

Example: https://feddit.de/post/794489
Does not appear in https://anonsys.net/profile/fussball although it's posted in that group.

### Describe alternatives you've considered

If a post with caption comes from a Lemmy or kbin account, it is a new post - this should generally be shared by groups.

**Comment by forenta** [comment]
Addition: The problem also occurs when you explicitly mention the group from Lemmy.
Post: https://feddit.de/post/821065
Group hiding it: https://anonsys.net/profile/caostest/conversations

**Comment by informapirata** [comment]
As I explained here https://github.com/LemmyNet/lemmy/issues/2707 the problem is not a problem that depends on Friendica, but derives from two distinct necessary conditions that allow Friendica to re-share messages addressed to the group/forum :
1) Friendica groups/forums re-share users' posts only if they are "followed" by those users
2) Friendica groups/forums re-share users' posts only if they are "mentioned" by those users

Unfortunately there are two types of problems:
1) a Lemmy user cannot follow another user and the Friendica groups/forums are users'; special users but still users
2) in Lemmy the "mentions" in the first message (message of type "page": with title and text) don't work; they are broken. They work in reply messages (post-type messages, the untitled ones, like mastodon messages).

When a Lemmy user "subscribes" to a Friendica group/forum, he can see it exactly as a Lemmy community (in fact, the Friendica forums and the Lemmy Communities are both ActivityPub "group" users), but:
1) the subscription made by a Lemmy user is not interpreted by Friendica as a follow (you can check among the followers of the Friendica group/forum
2) writing a new message made by a Lemmy user in a subscribed community is not interpreted by Friendica as a mention

Therefore the system does not work.

There must be two solutions and they concern Lemmy, but the Friendica developers could help those of Lemmy with ActivityPub.
In fact, the Friendica and Lemmy developers must agree to make the two systems talk. In particular:
1) Subscriptions by a Lemmy user to any ActivityPub "group" must be sent by Lemmy as a follow and accepted as such by Friendica. At most, the Friendica developers could try to make the "subscription" recognized as a follow (but in my opinion it is complicated and dangerous)
2) Lemmy developers need to solve the problem of mentions never working in the first opening message. At most, Friendica developers could try to make people recognize how a message written "inside" a "subscribed" community is interpreted as a mention (but in my opinion it's complicated and dangerous in this case too)

NB: in Kbin the mentions of the first message work, but the problem of the follow remains. Although Kbin users can also follow other users, subscribing to an activitypub "group" does not happen as a follow, exactly as in Lemmy. For this reason also Lemmy users cannot use Friendica groups/forums

PS: I would love to see this problem fixed too. Friendica could become the intersection between the lemmyverse and microblogging (mastodon, misskey, pleroma, etc), which in times of RedditMigration could be a nice added value.
To this end, I would also like the Friendica developers to be able to reproduce the trick that allows Kbin to make Mastodon users read their messages with the title correctly (cfr https://github.com/friendica/friendica/issues/13206)

**Comment by forenta** [comment]
> Although Kbin users can also follow other users, subscribing to an activitypub "group" does not happen as a follow, exactly as in Lemmy. For this reason also Lemmy users cannot use Friendica groups/forums

Lemmy user _do_ use Friendica groups and it's fine for them - the issue is only on 'our' side: We don't see what they post in 'our' groups. See https://feddit.de/c/fussball@anonsys.net

**Comment by informapirata** [comment]
> Lemmy user _do_ use Friendica groups and it's fine for them

No, they cannot use them fully: in fact a Lemmy user **cannot open a new thread** in a Friendica group/forum.

Fortunately with the latest updates a Lemmy user can reply in a thread re-shared by a Friendica group/forum. But the problem with the first post remains, due to the "broken" mentions of Lemmy

**Comment by forenta** [comment]
> in fact a Lemmy user cannot open a new thread in a Friendica group/forum

What makes you think that? The linked example shows the opposite.

On the contrary, it seems much more like using a Friendica group creates a parallel world: Friendica users continue to post things in it, Lemmy users post things in it - but neither sees the other's content. See:

https://anonsys.net/profile/fussball

vs.

https://feddit.de/c/fussball@anonsys.net

Same Friendica-group - both use it, separate content.

**Comment by tobiasd** [comment]
For me (Friendica 2023.09-dev), the postings to the Lemmy groups are displayed on the "Global Community" page but not when following the links in the "Group Widget" in my network stream.

**Comment by informapirata** [comment]
> See:
> 
> https://anonsys.net/profile/fussball
> 
> vs.
> 
> https://feddit.de/c/fussball@anonsys.net

Your example confirms my guess:

1) if you access from Lemmy, you will see the posts published by Lemmy users only because the Lemmy dB interprets the "Friendica forum" as if it were one of its own groups (=the Lemmy communities) and therefore incorporates those posts as it already knows how to do.. .

2) Friendica, on the other hand, will not take charge of those posts because one of the two necessary conditions that allow a "Friendica forum" (but also a GUP.PE group or a CHIRP group, because this is the correct way to do it in ActivityPub) is missing !) to re-share a post, i.e. the MENTION of the "Friendica forum"

**Comment by tobiasd** [comment]
> For me (Friendica 2023.09-dev), the postings to the Lemmy groups are displayed on the "Global Community" page but not when following the links in the "Group Widget" in my network stream.

Quick update here - for me with 2023.09-dev this now works.

**Comment by annando** [comment]
I invested quite some time now for the Lemmy interaction. The participation in Lemmy groups should work now fine. The other way (Lemmy users participation in Friendica groups) currently has got an issue that is to be fixed. There is an incompatibility between Friendica and Lemmy.
